diff --git a/js/app.js b/js/app.js
index 032d736..bd2d023 100644
--- a/js/app.js
+++ b/js/app.js
@@ -18,6 +18,7 @@
   var $clearCompleted = document.querySelector('.clear-completed');
   var $filters = document.querySelectorAll('.filters a');
   var escapePressed = false;
+  var savedEdit = false;
 
   function generateId() {
     return Date.now().toString(36) + Math.random().toString(36).substr(2, 5);
@@ -233,6 +234,7 @@
 
   function saveEdit(id, title) {
     title = title.trim();
+    savedEdit = true;
     if (!title) {
       deleteTodo(id);
       return;
@@ -299,8 +301,9 @@
 
   $todoList.addEventListener('blur', function (e) {
     if (e.target.classList.contains('edit')) {
-      if (escapePressed) {
+      if (escapePressed || savedEdit) {
         escapePressed = false;
+        savedEdit = false;
         return;
       }
       var li = e.target.closest('li');
